### PR TITLE
Fix 731 form validation

### PR DESCRIPTION
**Pull request by winged**
We should only validate answers that are visible.

### CODE REVIEW

**Comment by winged** [comment]
Fixes #731

File: caluma/form/tests/test_validators.py [hunk 1]
```diff
@@ -343,3 +343,24 @@ def test_validate_required_integer_0(
...
+    document = document_factory(form=form_question.form)
+
+    # answer is out of validation range so should fail
+    answer_factory(question=question, document=document, value=4)
+
+    if question.is_hidden == "true":
+        assert DocumentValidator().validate(document, info) is None
+    else:
+        with pytest.raises(ValidationError):
+            DocumentValidator().validate(document, info)
```
Comment (anehx) [inline caluma/form/tests/test_validators.py]: I think we should add a test for validating subforms when hidden...

Comment (winged) [reply caluma/form/tests/test_validators.py]: This does exactly that. Or do you mean yet another level of subforms?

Comment (winged) [reply caluma/form/tests/test_validators.py]: Nevermind :man_facepalming:

**Pull request merged**